pkg/ddevapp/config.go

```diff
@@ -309,6 +309,43 @@ func (app *DdevApp) WriteDockerComposeConfig() error {
 	return err
 }
 
+// CheckCustomConfig warns the user if any custom configuration files are in use.
+func (app *DdevApp) CheckCustomConfig() {
+
+	// Get the path to .ddev for the current app.
+	ddevDir := filepath.Dir(app.ConfigPath)
+
+	customConfig := false
+	if _, err := os.Stat(filepath.Join(ddevDir, "nginx-site.conf")); err == nil {
+		util.Warning("Using custom nginx configuration in nginx-site.conf")
+		customConfig = true
+	}
+
+	mysqlPath := filepath.Join(ddevDir, "mysql")
+	if _, err := os.Stat(mysqlPath); err == nil {
+		mysqlFiles, err := fileutil.ListFilesInDir(mysqlPath)
+		util.CheckErr(err)
+		if len(mysqlFiles) > 0 {
+			util.Warning("Using custom mysql configuration: %v", mysqlFiles)
+			customConfig = true
+		}
+	}
+
+	phpPath := filepath.Join(ddevDir, "php")
+	if _, err := os.Stat(phpPath); err == nil {
+		phpFiles, err := fileutil.ListFilesInDir(phpPath)
+		util.CheckErr(err)
+		if len(phpFiles) > 0 {
+			util.Warning("Using custom PHP configuration: %v", phpFiles)
+			customConfig = true
+		}
+	}
+	if customConfig {
+		util.Warning("Custom configuration takes effect after 'ddev restart'.")
```

Comment:
I don't think this is a valid message. These checks happen when the container is started, so it may be confusing to have a message telling you to restart while you're in the process of starting the container.

Reply:
No, unless I'm lost, these checks happen when `ddev start` is executed. That may or may not result in a container start. Until [ddev start should use docker-compose up --force-recreate](https://github.com/drud/ddev/issues/716)

Tell me if I'm just lost.